src/functions.php

```diff
@@ -212,30 +212,50 @@ function unwrap($promises)
  * respective positions to the original array. If any promise in the array
  * rejects, the returned promise is rejected with the rejection reason.
  *
- * @param mixed $promises Promises or values.
- * @param bool $recursive - If true, resolves new promises that might have been added to the stack during its own resolution.
+ * @param mixed $promises  Promises or values.
+ * @param bool  $recursive If true, resolves new promises that might have been added to the stack during its own resolution.
  *
  * @return PromiseInterface
  */
 function all($promises, $recursive = false)
+{
+    return all_limit($promises, null, $recursive);
+}
+
+/**
+ * Like all(), but only allows a certain number of outstanding promises at any
+ * given time.
+ *
+ * $concurrency may be an integer or a function that accepts the number of
+ * pending promises and returns a numeric concurrency limit value to allow for
+ * dynamic a concurrency size.
+ *
+ * @param mixed        $promises  Promises or values.
+ * @param int|callable $concurrency
```

Comment:
`$concurrency` can also be null (at least you pass null in `all_limit($promises, null, $recursive)`. so it should add the `|null` type.

Reply:
zero `0` seems to mean the same thing. So we could instead also remove the nullable as well.